### PR TITLE
Document Sigstore's graduated application to the OpenSSF

### PR DESCRIPTION
**Pull request by bobcallaway**
This represents a voting item for the Sigstore TSC, which is to formally document how the TSC believes we meet the criteria for an OpenSSF graduated project.

Since this is a voting item, it needs an explicit vote from all members of @sigstore/core-team.

Once approved, @lukehinds has agreed to convert this into the required documents and process as specified at https://github.com/ossf/tac/blob/main/process/project-lifecycle.md#graduated for discussion and vote at a future OpenSSF TAC meeting.

### CODE REVIEW

**Review by lukehinds — APPROVED** [review]

File: docs/best_practices_badge.md [hunk 1]
```diff
@@ -0,0 +1,741 @@
...
+
+### Identification
+
+#### What is the human-readable name of the project? 
+
+sigstore
+
+#### What is a brief description of the project? 
+
+Sigstore is an open source project for improving software supply chain security. The Sigstore framework and tooling empowers software developers and consumers to securely sign and verify software artifacts such as release files, container images, binaries, software bills of materials (SBOMs), and more. Signatures are generated with ephemeral signing keys so there’s no need to manage keys. Signing events are recorded in a tamper-resistant public log so software developers can audit signing events.
```
Comment (SantiagoTorres) [inline docs/best_practices_badge.md:15]: would it have been worth mentioning SLSA prov + in-toto attestations here?

**Review by SantiagoTorres — DISMISSED** [review]

**Review by priyawadhwa — DISMISSED** [review]
lgtm! thanks for writing this up @bobcallaway.

File: docs/best_practices_badge.md [hunk 2]
```diff
@@ -0,0 +1,741 @@
...
+
+#### It is SUGGESTED that the project implement continuous integration (where new or changed code is frequently integrated into a central code repository and automated tests are run on the result). 
+
+Each project within Sigstore is tested using idiomatic FLOSS testing tools for the respective language, and are regularly tested in GitHub Actions for each pull request.
+
+### New functionality testing
+
+#### The project MUST have a general policy (formal or not) that as major new functionality is added to the software produced by the project, tests of that functionality should be added to an automated test suite. 
+
+This is a general principle within sigstore, with evidence present across our repositories.
```
Comment (trevrosen) [inline docs/best_practices_badge.md:203]: Capitalize "Sigstore"

File: docs/best_practices_badge.md [hunk 3]
```diff
@@ -0,0 +1,741 @@
...
+
+#### The project MUST have evidence that the test\_policy for adding tests has been adhered to in the most recent major changes to the software produced by the project.
+
+This is a general principle within sigstore, with evidence present across our repositories.
```
Comment (trevrosen) [inline docs/best_practices_badge.md:207]: Capitalize "Sigstore"

File: docs/best_practices_badge.md [hunk 4]
```diff
@@ -0,0 +1,741 @@
...
+
+Our linting configs are stored in source control and can be audited for strictness. 
+
+## Security
+
+### Secure development knowledge
+
+#### The project MUST have at least one primary developer who knows how to design secure software. (See ‘details’ for the exact requirements.)
+
+Several of our maintainers work in security engineering departments (e.g. Google’s Open Source Security Team).
```
Comment (trevrosen) [inline docs/best_practices_badge.md:233]: Plz add "GitHub's Package Security Team" to the `e.g.` here 😃

File: docs/best_practices_badge.md [hunk 5]
```diff
@@ -0,0 +1,741 @@
...
+
+#### At least one of the project's primary developers MUST know of common kinds of errors that lead to vulnerabilities in this kind of software, as well as at least one method to counter or mitigate each of them. 
+
+Several of our maintainers work in security engineering departments (e.g. Google’s Open Source Security Team).
```
Comment (trevrosen) [inline docs/best_practices_badge.md:237]: See above 😄

File: docs/openssf_graduated_application.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+      - Maintainers from: Chainguard, Google, Trendyol 
+
+    - Sigstore-python: [Contributor data](https://github.com/sigstore/sigstore-python/graphs/contributors) 
+
+      - Maintainers from: Google, Trail of Bits, Stacklok
+
+    - Sigstore-rs: [Contributor data](https://github.com/sigstore/sigstore-rs/graphs/contributors)
+
+      - Maintainers from: Alibaba, Google, Red Hat, Stacklok, SUSE, Stacklok
```
Comment (trevrosen) [inline docs/openssf_graduated_application.md:55]: Stacklok is listed 2x

**Review by trevrosen — CHANGES_REQUESTED** [review]
I have just a few requested changes. Overall this looks great - thanks so much for taking the time here, @bobcallaway!

**Review by trevrosen — APPROVED** [review]

**Review by SantiagoTorres — APPROVED** [review]

**Review by lukehinds — APPROVED** [review]

**Pull request merged**